project/src/main/java/com/google/sps/servlets/RecommendationServlet.java

```diff
@@ -0,0 +1,75 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Restaurant;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/** Servlet that provides a restaurant recommendation. */
+@WebServlet("/recommendation")
+public class RecommendationServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+  private static final Logger LOGGER = Logger.getLogger(RecommendationServlet.class.getName());
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Set<Restaurant> restaurantSet = new HashSet<>();
+    BufferedReader reader = request.getReader();
+    String body = reader.readLine();
+    try {
+      addRestaurantsToSet(body, restaurantSet);
+    } catch (JSONException e) {
+      LOGGER.log(Level.WARNING, "Error parsing JSON: " + e.getMessage());
+    }
+    // Write one random restaurant to response. This will be replaced by the picking algorithm.
+    response.setContentType("application/json");
+    for (Restaurant restaurant : restaurantSet) {
+      response.getWriter().println(gson.toJson(restaurant));
+      return;
+    }
+  }
+
+  private void addRestaurantsToSet(String body, Set<Restaurant> restaurantSet)
```

Comment:
I think it would be worth writing a Restaurant static builder method that takes in a `JSONObject`/`Map` called `Restaurant fromJson(JSONObject json)` or something that will construct a Restaurant instance from JSON. This will cut down on most of this code here and move it into the Restaurant class where it makes more sense, since this is really constructor-esque code inside this method.